pkgs/development/python-modules/pydantic/default.nix

```
@@ -43,6 +69,18 @@ buildPythonPackage rec {
     export HOME=$(mktemp -d)
   '';
 
+  # Must include current directory into PYTHONPATH, since documentation
+  # building process expects "import pydantic" to work.
+  preBuild = ''
+    PYTHONPATH=$PWD:$PYTHONPATH make docs
+  '';
+
+  # Layout documentation in same way as "sphinxHook" does.
+  postInstall = ''
+    mkdir -p $out/share/doc/$name
+    mv ./site $out/share/doc/$name/html
```

```suggestion
    install -D ./site $out/share/doc/${pname}/html
```
Having version in there is an artifact of package managers like apt. We don't need that.

---

Quite the opposite. It does not make sense to install documentation in package `foo-doc`
into `/usr/share/doc/foo-$version` with dpkg because you won't be able to install multiple version of `foo-doc` binary package at same time anyway. With Nix you can.